modules/ismp/pallet/pallet-asset-transfer/src/lib.rs

```diff
@@ -0,0 +1,419 @@
+        DispatchPostError,
+    }
+
+    /// Events emiited by the relayer pallet
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// An XCM transfer from the relay chain has been transformed into a crosschain message
+        TransferInitiated {
+            /// Source account on the relaychain
+            from: T::AccountId,
+            /// beneficiary account on destination
+            to: T::EvmAccountId,
+            /// Amount transferred
+            amount: <T::Assets as fungibles::Inspect<T::AccountId>>::Balance,
+            /// Destination chain
+            dest: StateMachine,
+        },
+
+        /// An asset has been received and transferred to the beneficiary's account on the
+        /// relaychain
+        AssetReceived {
+            /// beneficiary account on relaychain
+            beneficiary: T::AccountId,
+            /// Amount transferred
+            amount: <T::Assets as fungibles::Inspect<T::AccountId>>::Balance,
+            /// Destination chain
+            source: StateMachine,
+        },
+    }
+}
+
+impl<T: Config> Pallet<T>
+where
+    u128: From<<T::Assets as fungibles::Inspect<T::AccountId>>::Balance>,
+    T::AccountId: Into<[u8; 32]>,
+    T::EvmAccountId: Into<[u8; 20]>,
+{
+    pub fn account_id() -> T::AccountId {
+        T::PalletId::get().into_account_truncating()
+    }
+
+    pub fn protocol_account_id() -> T::AccountId {
+        T::ProtocolAccount::get().into_account_truncating()
+    }
+
+    /// Dispatch ismp request to token gateway on destination chain
+    pub fn dispatch_request(
+        multi_account: MultiAccount<T::AccountId, T::EvmAccountId>,
+        amount: <T::Assets as fungibles::Inspect<T::AccountId>>::Balance,
+    ) -> Result<(), Error<T>> {
+        let dispatcher = Dispatcher::<T>::default();
+
+        let mut to = [0u8; 32];
+        let _temp: [u8; 20] = multi_account.evm_account.clone().into();
+        to.iter_mut().enumerate().for_each(|(index, item)| {
+            if index < 20 {
+                *item = _temp[index];
+            }
+        });
+        let from: [u8; 32] = multi_account.substrate_account.clone().into();
+        let asset_id = T::DotAssetId::get().0.into();
+        let body = Body {
+            amount: {
+                let amount: u128 = amount.into();
+                let mut bytes = [0u8; 32];
+                U256::from(amount).to_big_endian(&mut bytes);
+                alloy_primitives::U256::from_be_bytes(bytes)
+            },
+            asset_id,
+            redeem: false,
+            from: from.into(),
+            to: to.into(),
+        };
+
+        let dispatch_post = DispatchPost {
+            dest: multi_account.dest_state_machine,
+            from: T::TokenGateWay::get().0.to_vec(),
+            to: T::TokenGateWay::get().0.to_vec(),
+            // 1 hour timeout
+            timeout_timestamp: 60 * 60,
```

Comment:
Is there any way to make this configurable from the xcm call?

Reply:
Yeah, I can probably use the generalIndex variant to accept the timeout